tests/test_deprecations.py

```diff
@@ -18,8 +22,9 @@
     StreamingSequentialStrategy,
     TwoPhaseStrategy,
 )
-from baybe.targets import Objective
+from baybe.targets import Objective as ObjectiveFromTargets
```

Comment:
that is a deprecation that id be willing to drop in this PR, its been some time

Reply:
I think it's time to finally agree on a deprecation protocol in our next dev meeting and then simply stick to it instead of deciding case by case. Something like:
1. Keep the deprecation for X versions (changelog: "deprecated")
2. Expire the deprecation of Y versions, meaning we turn the warning into an actual error (changelog: "expired deprecations")
3. Remove the deprecation entirely (changelog: "removed deprecations")

... or similar. Shall we just postpone the discussion until the next meeting?

Reply:
are ou seriously suggesting this 3 step process without any automation?
imo this is one step too complicated

1) deprecate
2) expire and remove deprecation in 1 step
imo is enough

this also doesnt solve the unanswered question: how long in terms of time between 1 and 2? Unless there is a choice and a way to track / automate that choice we have no other option than case-by-case (fine for me)

Reply:
Ok, we've agreed on the policy in the dev meeting, so I'll close this one. Can prepare the necessary deprecation PR for the next release after merging this one 👍🏼